nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -595,36 +638,178 @@ sub multiLineList {
   # services.openssh.enable = true;
 
   # Open ports in the firewall.
+
   # networking.firewall.allowedTCPPorts = [ ... ];
   # networking.firewall.allowedUDPPorts = [ ... ];
-  # Or disable the firewall altogether.
+
+  # WARNING! WARNING! WARNING! INSECURE!
+  #
+  # To disable the firewall altogether use the following.
+  #
+  # Use only inside trusted networks behind another firewall or else
+  # bad people across the galaxy will be able to access any ports you
+  # leave open.
+  #
   # networking.firewall.enable = false;
 
   # Enable CUPS to print documents.
   # services.printing.enable = true;
 
-  # Enable sound.
+  # Enable ALSA. This option will add ALSA tools to your
+  # systemPackages and make ALSA settings such as volume persistent
+  # across reboots.
+  # Not enabling this doesn't actually disable sound output via ALSA
+  # because ALSA is the default sound API of the Linux kernel.
+
   # sound.enable = true;
+
+  # Configure ALSA sound volume and capture control via common media
+  # keys. This will have no effect unless `sound.enable` is set.
+  # You want to leave this disabled if you run a desktop environment
+  # like KDE, Gnome, Xfce, etc, as those handle such things
+  # themselves.
+  # You might want to enable this if you run a minimalistic desktop
+  # environment or work from bare linux ttys/framebuffers.
+
+  # sound.mediaKeys.enable = true;
```

Comment:
I think removing this section about mediakeys would make it easier to merge, since mediakeys being an option is a bit contentious.